client/packages/common/src/intl/strings/IntlStrings.ts

```diff
@@ -31,9 +31,9 @@ export const useTranslation = (ns?: Namespace): TypedTFunction<LocaleKey> => {
           ? { ...opts, returnDetails: false }
           : { returnDetails: false, defaultValue };
 
-      return key
+      return key && i18n.exists(key, options)
         ? t(key, options)
-        : String(defaultValue || options?.defaultValue || '');
+        : String(defaultValue || options?.defaultValue || key || '');
```

Comment:
Ohh - open to feedback on this one... was a side quest. We get lots of warnings in console when working with JSON forms - seeing as they are custom data, we don't translate the strings, was clogging console. Potentially this should be a flag passed into useTranslation from JSON Form components though, so we still get warnings otherwise? 
![Screenshot 2025-06-10 at 9 16 02 AM](https://github.com/user-attachments/assets/2d005994-e1c4-48e4-a783-a5395019c5ee)

Idk, personally haven't found the logs helpful, if I'm working on a page, I'll see the missing translation key in the UI before I see the log 👀

Reply:
GOOD